### PR TITLE
[AIC-py][eval] bundle of small easy fixes

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[AIC-py][eval] bundle of small easy fixes

- freeze dataclasses
- wire up eval fn timeout further (there is a little more to go)
- move mock aiconfig runtime to separate file
-

see https://docs.google.com/document/d/15gx4vVmKVgI5bDHjSDd8xCQmTGcC8RrtLU0iCa32AIQ/edit

### CODE REVIEW

File: python/src/aiconfig/eval/lib.py [hunk 1]
```diff
@@ -90,7 +91,7 @@ def __str__(self) -> str:
 
 
 async def _evaluate_for_sample(
-    eval_params: SampleEvaluationParams[T_InputDatum, T_OutputDatum, T_MetricValue]
+    eval_params: SampleEvaluationParams[T_InputDatum, T_OutputDatum, T_MetricValue], timeout: int
```
Comment (saqadri) [inline python/src/aiconfig/eval/lib.py:94]: would recommend renaming the variable to include the time unit. For e.g. `timeout_ms`

File: python/src/aiconfig/eval/lib.py [hunk 2]
```diff
@@ -123,9 +123,9 @@ def _ok_with_log(res_: Result[T_MetricValue, str]) -> T_MetricValue | None:
 
 
 async def evaluate(
-    evaluation_params_list: DatasetEvaluationParams[T_InputDatum, T_OutputDatum],
+    evaluation_params_list: DatasetEvaluationParams[T_InputDatum, T_OutputDatum], eval_fn_timeout: int
```
Comment (saqadri) [inline python/src/aiconfig/eval/lib.py:126]: Same as above. `eval_fn_timeout_ms`

**Review by saqadri — APPROVED** [review]
LGTM, just a few comments

**Comment by rholinshead** [comment]
For the description, can you update to state what the fixes are here? That will be easier for changelogs and ensures anyone can see the description in the public repo (that doc isn't public)

File: python/src/aiconfig/eval/lib.py (hunk 1 above)
Comment (jonathanlastmileai) [reply python/src/aiconfig/eval/lib.py:94]: Added here https://docs.google.com/document/d/15gx4vVmKVgI5bDHjSDd8xCQmTGcC8RrtLU0iCa32AIQ/edit

**Pull request merged**